src/ripple/shamap/impl/SHAMap.cpp

```diff
@@ -164,7 +164,7 @@ SHAMapTreeNode* SHAMap::walkToPointer (uint256 const& id) const
     }
 
     auto ret = static_cast<SHAMapTreeNode*>(inNode);
```

Comment:
The static type of `inNode` is `SHAMapAbstractNode` as it can be a leaf node or an inner node at run time.  An invariant of the above loop is that when it exits, `inNode` is not an inner node, therefore it must be a leaf node.  The `static_cast` converts the `SHAMapAbstractNode*` (static type) to `SHAMapTreeNode*` (the known dynamic type).